### PR TITLE
Small change for RISCV compilation

### PR DESCRIPTION
**Pull request by puma314**

### CODE REVIEW

File: secp256k1-sys/build.rs [hunk 1]
```diff
@@ -31,6 +31,38 @@ fn main() {
...
+            println!("cargo:warning=Variable RISCV_GNU_TOOLCHAIN unset. Assuming '{DEFAULT_RISCV_GNU_TOOLCHAIN}'");
+            println!("cargo:warning=Please make sure to build riscv toolchain:");
+            println!("cargo:warning=  git clone https://github.com/riscv-collab/riscv-gnu-toolchain && cd riscv-gnu-toolchain");
+            println!("cargo:warning=  export RISCV_GNU_TOOLCHAIN={DEFAULT_RISCV_GNU_TOOLCHAIN}");
+            println!("cargo:warning=  configure --prefix=\"$RISCV_GNU_TOOLCHAIN\" --with-arch=rv32im --with-abi=ilp32");
+            println!("cargo:warning=  make -j$(nproc)");
+
+            // if unset, try the default and fail eventually
+            DEFAULT_RISCV_GNU_TOOLCHAIN.into()
+        });
```
Comment (Kixunil) [inline secp256k1-sys/build.rs:48]: I don't think it's appropriate to force every RISCV compilation to use the GNU toolchains - are there not other toolchains?

File: secp256k1-sys/build.rs [hunk 2]
```diff
@@ -31,6 +31,38 @@ fn main() {
...
+
+        base_config
+            .compiler("clang")
+            .no_default_flags(true)
+            .flag(&format!("--sysroot={riscv_gnu_toolchain_path}/riscv32-unknown-elf"))
```
Comment (Kixunil) [inline secp256k1-sys/build.rs:53]: IIRC this is beyond our MSRV.

**Review by Kixunil — CHANGES_REQUESTED** [review]
Please provide more details. Why is the change needed, what problem does it solve and why you picked this particular solution?

**Comment by apoelstra** [comment]
Agreed, I'd like a lot more details and a CI job that compiles with RISCV.

If we currently can't compile with riscv and now we can, that's cool, and I'd be willing to compromise on MSRV or other forms of flexibility to get that. But from this PR I can't tell what's going on.

**Pull request closed without merging**